yolo_detector:
  ros__parameters:
    # Model configuration
    model_path: "yolov8_det_float.tflite"
    confidence_threshold: 0.5
    nms_threshold: 0.4
    
    # Input image dimensions
    input_width: 640
    input_height: 640
    
    # Topics
    input_image_topic: "/camera/image_raw"
    output_detections_topic: "/ai_inference/detections"
    
    # Performance settings
    max_detections: 100
    use_gpu: true
    
    # COCO class names (80 classes)
    class_names:
      - "person"
      - "bicycle"
      - "car"
      - "motorcycle"
      - "airplane"
      - "bus"
      - "train"
      - "truck"
      - "boat"
      - "traffic light"
      - "fire hydrant"
      - "stop sign"
      - "parking meter"
      - "bench"
      - "bird"
      - "cat"
      - "dog"
      - "horse"
      - "sheep"
      - "cow"
      - "elephant"
      - "bear"
      - "zebra"
      - "giraffe"
      - "backpack"
      - "umbrella"
      - "handbag"
      - "tie"
      - "suitcase"
      - "frisbee"
      - "skis"
      - "snowboard"
      - "sports ball"
      - "kite"
      - "baseball bat"
      - "baseball glove"
      - "skateboard"
      - "surfboard"
      - "tennis racket"
      - "bottle"
      - "wine glass"
      - "cup"
      - "fork"
      - "knife"
      - "spoon"
      - "bowl"
      - "banana"
      - "apple"
      - "sandwich"
      - "orange"
      - "broccoli"
      - "carrot"
      - "hot dog"
      - "pizza"
      - "donut"
      - "cake"
      - "chair"
      - "couch"
      - "potted plant"
      - "bed"
      - "dining table"
      - "toilet"
      - "tv"
      - "laptop"
      - "mouse"
      - "remote"
      - "keyboard"
      - "cell phone"
      - "microwave"
      - "oven"
      - "toaster"
      - "sink"
      - "refrigerator"
      - "book"
      - "clock"
      - "vase"
      - "scissors"
      - "teddy bear"
      - "hair drier"
      - "toothbrush"
